### PR TITLE
fix(Dataspreadsheet): add callback function for after drag/drop ends

### PR DESCRIPTION
**Pull request by AustinGitHub**
Closes https://github.com/carbon-design-system/ibm-products/issues/5620

#### What did you change?

I added a callback method so user can know what new column headers are after drag/drop is done

#### How did you test and verify your work?

Storybook

https://github.com/carbon-design-system/ibm-products/assets/10100397/00c0e6d4-42de-4b5d-aaed-fb34c6bc7b99

### CODE REVIEW

**Review by davidmenendez — CHANGES_REQUESTED** [review]
code looks good. only thing i'd like to see changed is to rename the prop to something like `onColDrag` or something just to continue to maintain the `on` naming convention for handlers.

**Comment by AustinGitHub** [comment]
Added the fix, thanks!

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx [hunk 1]
```diff
@@ -285,6 +294,46 @@ export let DataSpreadsheet = React.forwardRef(
...
+      }
+    }, [
+      previousState?.selectedHeaderReorderActive,
+      currentColumns,
+      headerCellHoldActive,
+      columns,
+      activeCellContent,
+      onColDrag,
+    ]);
+
```
Comment (amal-k-joy) [inline packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:336]: Hey @AustinGitHub ,
Code looks fine, but I was thinking of this scenario. 
Let initial order of columns be `orderDefault`. 
When colums are dragged , order changed from `orderDefault` to `order 1` , then changed to `order 2`. Then again dragged back to `orderDefault`. At this last drag scenario , will this callback get triggers?
@davidmenendez Can you share your thoughts? Is this a required scenario ?

**Review by amal-k-joy — CHANGES_REQUESTED** [review]

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx (hunk 1 above)
Comment (AustinGitHub) [reply packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:336]: Let me check

Comment (AustinGitHub) [reply packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:336]: You're right, it doesn't get triggered in this use case. I will add a fix for this

**Comment by AustinGitHub** [comment]
https://github.com/carbon-design-system/ibm-products/assets/10100397/7624e79a-3189-4b0f-9b3c-dac3199671e2

Here is the fix

**Review by davidmenendez — APPROVED** [review]
@amal-k-joy good catch!

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx [hunk 2]
```diff
@@ -296,34 +297,41 @@ export let DataSpreadsheet = React.forwardRef(
...
 
       if (
         !previousState.selectedHeaderReorderActive &&
         !headerCellHoldActive &&
         currentHeaders.length > 0 &&
-        pastHeaders.length > 0 &&
-        JSON.stringify(currentHeaders) !== JSON.stringify(pastHeaders)
+        pastColumns.length > 0 &&
+        JSON.stringify(currentHeaders) !== JSON.stringify(pastColumns)
       ) {
```
Comment (amal-k-joy) [inline packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:318]: Hey @AustinGitHub ,
Thanks for the changes and those looks good. But I am seeing many duplicate lines of code in the `if` block and `else if` block. Is it possible to do some refactor?

**Review by amal-k-joy — CHANGES_REQUESTED** [review]

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx (hunk 2 above)
Comment (AustinGitHub) [reply packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:318]: Sure

Comment (AustinGitHub) [reply packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx]: I fixed it, thanks!

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx [hunk 3]
```diff
@@ -285,6 +295,50 @@ export let DataSpreadsheet = React.forwardRef(
...
+        setPastColumns(currentHeaders);
+      }
+
+      if (
+        !previousState.selectedHeaderReorderActive &&
+        pastColumns.length > 0 &&
+        !headerCellHoldActive &&
+        JSON.stringify(currentHeaders) !== JSON.stringify(pastColumns)
+      ) {
+        if (currentHeaders.length > 0) {
```
Comment (amal-k-joy) [inline packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:318]: Hey, can you check if we can modify line 318 to 330 as below block ?
```
 onColDrag({
    headers: currentHeaders,
    data: activeCellContent.props.data,
  });
  if (currentHeaders.length === 0){
    setPastColumns([]);
  }
```

**Review by amal-k-joy — CHANGES_REQUESTED** [review]

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx (hunk 3 above)
Comment (AustinGitHub) [reply packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:318]: Sure, checking

Comment (AustinGitHub) [reply packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx]: This works, thanks, just pushed the code change

**Review by amal-k-joy — APPROVED** [review]
Nice work Austin

**Review by davidmenendez — APPROVED** [review]

**Pull request merged**